handler/handler.go

```diff
@@ -0,0 +1,145 @@
+package handler
+
+import (
+	"encoding/json"
+	"fmt"
+	"main/models"
+	"net/http"
+	"time"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+)
+
+type Service interface {
+	CreateRecord(request models.TransformRequest) (*models.Record, error)
+	GetRecord(id string) (*models.Record, error)
+	GetAllRecords() ([]models.Record, error)
+	UpdateRecord(id string, request models.TransformRequest) *models.Record
+	DeleteRecord(id string) error
+}
+
+type Handler struct {
+	service Service
+}
+
+func NewHandler(service Service) *Handler {
+	return &Handler{
+		service: service,
+	}
+}
+
+func (h *Handler) RunServer() error {
+	router := chi.NewRouter()
+	router.Use(middleware.Logger)
+	router.Use(middleware.Recoverer)
```

Comment:
use `recoverer` middleware first to handle all possible panics, even in `logger` middleware